### PR TITLE
Add support for Norges Bank API for currencies

### PR DESCRIPTION
**Pull request by ohjakobsen**
Adds functionality to fetch exchange rates for international currencies against the Norwegian krone with the `get_exchange_rate()` function. This pull request solves #4

### CODE REVIEW

File: R/norges-bank.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    end = end, dynamic = dynamic, n_obs = n_obs, raw_response = raw_response
+  )
+  if (raw_response) return(invisible(dl))
+  dl <- do.call(rbind, dl)
+  dl
+}
+
+#' get_exchange_rate_
+#' @noRd
+get_exchange_rate_ <- function(currency, frequency, dynamic, n_obs, start,
```
Comment (Aeilert) [inline R/norges-bank.R:59]: I ended up using `*_single` for these functions. Happy with either option, but great if we could standardise.

File: R/norges-bank-utils.R [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+#' request_sdmx
+#' @noRd
+request_sdmx <- function(currency, frequency, start, end) {
+  api_path <- sprintf(
+    'data/EXR/%s.%s.NOK.SP?format=sdmx-json&startPeriod=%s&endPeriod=%s&locale=en',
+    frequency, currency, start, end
+  )
+  req <- request(nbank_url) |> req_url_path_append(api_path)
```
Comment (Aeilert) [inline R/norges-bank-utils.R:19]: A minor style question: Should piping always be done with a new line? That is: 

```r
request(nbank_url) |> 
  req_url_path_append(api_path)
```

File: tests/testthat/test-norges-bank.R [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+})
+
+test_that('get_exchange_rate() works when raw_response = TRUE', {
+  res <- get_exchange_rate('USD', raw_response = TRUE)
+  expect_true(is.list(res))
+  expect_true(is.list(res[[1]]))
+  expect_equal(names(res[[1]]), c('url', 'status', 'content', 'response'))
+  expect_identical(class(res[[1]]), 'noAPI')
+  expect_true(is.list(res[[1]]$content))
+})
```
Comment (Aeilert) [inline tests/testthat/test-norges-bank.R:44]: Consider adding a test for invisible response? ```expect_invisible(get_exchange_rate(...)```

**Review by Aeilert — COMMENTED** [review]
This looks good to me! Just some minor comments. 

Also; Remember to add your exported function to zzz.R so that it is also included in the caching.

File: R/norges-bank.R (hunk 1 above)
Comment (ohjakobsen) [reply R/norges-bank.R:59]: I agree to standardising on `*_single`, as it's also descriptive of what the function does.

File: R/norges-bank-utils.R (hunk 2 above)
Comment (ohjakobsen) [reply R/norges-bank-utils.R:19]: I'm inclined to agree on this one, although I do like onliners 😅

File: tests/testthat/test-norges-bank.R (hunk 3 above)
Comment (ohjakobsen) [reply tests/testthat/test-norges-bank.R:44]: Good idea!

**Comment by ohjakobsen** [comment]
The issues raised above should be resolved with new commit. I have chosen _not_ to include the function to the package cache. The reason behind this, is that data from the API is updated daily (at about 4 PM). With the default settings of the function, the result will become obsolete in a long running R process with the caching level set. Both the request and the response is quite lightweight, so local caching should be less needed.

**Comment by Aeilert** [comment]
> The issues raised above should be resolved with new commit. I have chosen _not_ to include the function to the package cache. The reason behind this, is that data from the API is updated daily (at about 4 PM). With the default settings of the function, the result will become obsolete in a long running R process with the caching level set. Both the request and the response is quite lightweight, so local caching should be less needed.

Got it. Fully agree with your decision then.

**Pull request merged**